### PR TITLE
[xcode16] Use [Experimental] instead of [RequiresPreviewFeatures].

### PR DESCRIPTION
**Pull request by rolfbjarne**
This is because `EnablePreviewFeatures=true` doesn't quite work, since it
requires the building .NET and the target .NET to be on the same version.
We're might want to build with .NET 9, but the Xcode branch is targeting .NET
8, so it doesn't work.

This behavior is explained here:
https://github.com/dotnet/designs/blob/main/accepted/2021/preview-features/preview-features.md#meaning-of-property-in-multi-targeted-projects

The best solution seems to switch to using the Experimental attribute instead,
which was designed for our scenario (and explicitly to fix the problem we're
running into): https://github.com/dotnet/designs/blob/bba3216250cb29b0063bac3ebb57a542ee21ad4f/accepted/2023/preview-apis/preview-apis.md

This also meant we had to augment `-nowarn` for bgen to:

* Pass any nowarn values to the compiler when bgen compiles stuff.
* Pass `$(NoWarn)` (the MSBuild property) to bgen when building a binding project.

### CODE REVIEW

File: docs/preview-apis.md [hunk 1]
```diff
@@ -85,6 +94,6 @@ framework as preview API for a while.
 
 We've tentatively set .NET 11 as the release when we'll stop marking FSKit as preview API.
 
-The diagnostic id for CryptoTokenKit is APL0001.
```
Comment (dalexsoto) [inline docs/preview-apis.md:88]: What happened to CryptoTokenKit?

File: tests/common/Configuration.cs [hunk 2]
```diff
@@ -310,6 +311,7 @@ static Configuration ()
 			DotNetTfm = GetVariable ("DOTNET_TFM", null);
 			iOSSupports32BitArchitectures = !string.IsNullOrEmpty (GetVariable ("IOS_SUPPORTS_32BIT_ARCHITECTURES", ""));
 			EnableXamarin = !string.IsNullOrEmpty (GetVariable ("ENABLE_XAMARIN", ""));
+			XcodeIsStable = string.Equals (GetVariable ("ENABLE_XAMARIN", ""), "true", StringComparison.OrdinalIgnoreCase);
```
Comment (dalexsoto) [inline tests/common/Configuration.cs:314]:
```suggestion
			XcodeIsStable = string.Equals (GetVariable ("XCODE_IS_STABLE", ""), "true", StringComparison.OrdinalIgnoreCase);
```

**Review by dalexsoto — COMMENTED** [review]
Looking good!

File: docs/preview-apis.md (hunk 1 above)
Comment (rolfbjarne) [reply docs/preview-apis.md:88]: Nothing, this was a copy/paste mistake when the section about FSKit was added.

**Review by mcumming — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by haritha-mohan — APPROVED** [review]

**Review by emaf — APPROVED** [review]

**Pull request merged**